File: ValidParentheses/ValidPar.java
public class ValidPar {
    public boolean isValid(String s) {
        Stack stack = new Stack();
        for(int i = 0; i < s.length();i ++){
            if(s.charAt(i) == '(' || s.charAt(i) == '{' ||s.charAt(i) == '[')
                stack.push(s.charAt(i));
            else{
                String temp = stack.empty()? null : stack.pop().toString();// if nly right para are in the string
                temp = temp + s.charAt(i);
                if(!temp.equals("()") && !temp.equals("[]") && !temp.equals("{}"))
                    return false;
            }
        }
        return stack.empty()?true:false;
    }
}

File: ValidParentheses/ValidPar.rb
# @param {String} s
# @return {Boolean}
def is_valid(s)
    st = []
    line = s.split("")
    line.each{|x|
        if (x =~ /[{\(\[]/) != nil
            st.push x
        else
            if (st.empty?) || (((st.pop + x) =~ /^(\(\))?({})?(\[\])?$/) == nil)# without ^ and $ would read empty string
                return false
            end
        end
    }
    st.empty?
end